From aea57b724a20ab788bc9c755dc080c3346bc983c Mon Sep 17 00:00:00 2001
From: tonishi92 <tatsuo.onishi@gmail.com>
Date: Tue, 27 Aug 2024 14:08:48 +0200
Subject: [PATCH 02/55] (Update) Correction to
 /noahmp/src/module_sf_noahmplsm.F

Add the following two lines before subroutine SFCDIF2 for OPT_SFC == 2 case.

CH = CH * UR
CM = CM * UR

There are two occurrences.

As noahmp is maintained under another git,
no patch from WRF git repository can not be used to modify it.
To work around this problem, we modify Makefile so that
"git am -3 <patch>" command be applied in Makefile in /phys.
---
 ...hys-noahmp-src-module_sf_noahmplsm.F.patch | 44 +++++++++++++++++++
 phys/Makefile                                 |  1 +
 2 files changed, 45 insertions(+)
 create mode 100644 phys/CORRECTION-phys-noahmp-src-module_sf_noahmplsm.F.patch

diff --git a/phys/CORRECTION-phys-noahmp-src-module_sf_noahmplsm.F.patch b/phys/CORRECTION-phys-noahmp-src-module_sf_noahmplsm.F.patch
new file mode 100644
index 00000000..e9d40161
--- /dev/null
+++ b/phys/CORRECTION-phys-noahmp-src-module_sf_noahmplsm.F.patch
@@ -0,0 +1,44 @@
+From 10403cb985208568d08bd6da205d9b880028b198 Mon Sep 17 00:00:00 2001
+From: tonishi92 <tatsuo.onishi@gmail.com>
+Date: Mon, 26 Aug 2024 16:39:27 +0200
+Subject: [PATCH] (CORRECTION) phys/noahmp/src/module_sf_noahmplsm.F
+ --------------  ! CORRECTION  CH = CH * UR  CM = CM * UR  ! CORRECTION
+ -------------- before "CALL SFCDIF2"
+
+---
+ src/module_sf_noahmplsm.F | 12 ++++++++++--
+ 1 file changed, 10 insertions(+), 2 deletions(-)
+
+diff --git a/src/module_sf_noahmplsm.F b/src/module_sf_noahmplsm.F
+index 3ec4ade..cda6da7 100644
+--- a/src/module_sf_noahmplsm.F
++++ b/src/module_sf_noahmplsm.F
+@@ -3894,7 +3894,11 @@ ENDIF   ! CROPTYPE == 0
+                        CM     ,CH     ,FV     ,CH2     )          !out
+        ENDIF
+      
+-       IF(OPT_SFC == 2) THEN
++       IF(OPT_SFC == 2) THEN 
++           ! CORRECTION 
++           CH = CH * UR 
++           CM = CM * UR 
++           ! CORRECTION 
+           CALL SFCDIF2(parameters,ITER   ,Z0M    ,TAH    ,THAIR  ,UR     , & !in
+                        ZLVL   ,ILOC   ,JLOC   ,         & !in
+                        CM     ,CH     ,MOZ    ,WSTAR  ,         & !in
+@@ -4364,7 +4368,11 @@ ENDIF   ! CROPTYPE == 0
+                        CM     ,CH     ,FV     ,CH2     )          !out
+         ENDIF
+ 
+-        IF(OPT_SFC == 2) THEN
++        IF(OPT_SFC == 2) THEN 
++           ! CORRECTION 
++           CH = CH * UR 
++           CM = CM * UR 
++           ! CORRECTION 
+           CALL SFCDIF2(parameters,ITER   ,Z0M    ,TGB    ,THAIR  ,UR     , & !in
+                        ZLVL   ,ILOC   ,JLOC   ,         & !in
+                        CM     ,CH     ,MOZ    ,WSTAR  ,         & !in
+-- 
+2.31.1
+
diff --git a/phys/Makefile b/phys/Makefile
index a7fb3daf..e26f403e 100644
--- a/phys/Makefile
+++ b/phys/Makefile
@@ -262,6 +262,7 @@ submodules :
 	      \( ! -f module_sf_noahmp_groundwater.F \) -o \( ! -f module_sf_noahmplsm.F \) ] ; then \
 		echo Pulling in NoahMP submodule ; \
 		( cd .. ; git submodule update --init --recursive ) ; \
+                ( cd noahmp ; git am -3 ../CORRECTION-phys-noahmp-src-module_sf_noahmplsm.F.patch ) ; \
 		ln -sf noahmp/drivers/wrf/module_sf_noahmpdrv.F . ; \
 		ln -sf noahmp/src/module_sf_noahmp_glacier.F . ; \
 		ln -sf noahmp/src/module_sf_noahmp_groundwater.F . ; \
-- 
2.31.1

